src/repository.c

```diff
@@ -56,6 +56,7 @@ static const struct {
 	{ GIT_REPOSITORY_ITEM_COMMONDIR, GIT_REPOSITORY_ITEM_GITDIR, "packed-refs", false },
 	{ GIT_REPOSITORY_ITEM_COMMONDIR, GIT_REPOSITORY_ITEM_GITDIR, "remotes", true },
 	{ GIT_REPOSITORY_ITEM_COMMONDIR, GIT_REPOSITORY_ITEM_GITDIR, "config", false },
+	{ GIT_REPOSITORY_ITEM_COMMONDIR, GIT_REPOSITORY_ITEM_GITDIR, "config.worktree", false },
```

Comment:
I am unsure about this definition (and don't yet know how to functionally test my changes outside of the unit test I introduced in the prior commit).

Reply:
Resolving this after the introduction of other unit tests and verifying behavior with the debugger.